code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -913,7 +913,7 @@
   * * AI - AI unit that initiated the deployment into the AI shell
   */
 /mob/living/silicon/robot/proc/deploy_init(mob/living/silicon/ai/AI)
-	real_name = "[AI.real_name] Shell-[ident]"
+	real_name = "[AI.real_name] [designation] Shell-[ident]"
```

Comment:
```suggestion
	real_name = "[AI.real_name] Shell-[uppertext(copytext(designation, 1, 4))][ident]"
```
This will add abberviated designation from first three letters of the module, like ENG for Engineering or MED for Medical infront of an ident. This way, name length will stay sane and AIs and everyone else will be able to discern module of the shell by glance. I believe you would agree we both wasted enough time on this, accept this suggestion and I have no further objections and support this PR.

Reply:
The image above was a maximum extreme, I have never seen AI name that long ingame honestly. Box could be made larger in width, but that would mean to make it so for several interfaces, and it would be required to be them be almost as big as a game window itself, just for one edge case, is rather ehh. With only ENG used, this would not really happen ingame, as even for AIs with a longer names this would be enough to make it fit.

Reply:
it is not an "edge case" ask to have interfaces support the maximum allowable name length with the maximum combination of module number (AKA Engineering Shell-XX).

Regardless though that ask would not be on this PR anyways.